### PR TITLE
State ws

### PR DESCRIPTION
**Pull request by Sarahtonein**
**Example:**

**Sent to playerservice:**

```
[
{
	"ID": 5,
	"stateID": 67,
	"value": 0.73
},
{
		"ID": 6,
		"stateID": 70,
		"value": 0.11
	}
]
```

**Received from playerservice:**

 ```
[
	{
		"ID": 5,
		"CreatedAt": "2023-11-06T03:35:24.467926031Z",
		"UpdatedAt": "2023-11-09T02:28:18.293881202Z",
		"DeletedAt": null,
		"PlayerID": 9,
		"stateID": 67,
		"value": 0.73
	},
	{
		"ID": 6,
		"CreatedAt": "2023-11-06T03:37:31.050496586Z",
		"UpdatedAt": "2023-11-09T02:28:18.340899983Z",
		"DeletedAt": null,
		"PlayerID": 10,
		"stateID": 70,
		"value": 0.11
	}
]

```

**Related Issue:**

https://github.com/BloomGameStudio/PlayerService/issues/67

### CODE REVIEW

File: handlers/state.go
Comment (Balugrizzly) [inline handlers/state.go:1]: Why does this use an Array function signature?
All other handlers don't

**Review by Balugrizzly — CHANGES_REQUESTED** [review]
Consider Using the ErrorHandler Functions

Can you Trim down the log Statements ?

File: handlers/state.go
Comment (Sarahtonein) [reply handlers/state.go:1]: > Why does this use an Array function signature? All other handlers don't

I thought we were taking in an array of states for some reason. Can change this easily if that is not the case.

File: controllers/ws/state/stateWriter.go [hunk 1]
```diff
@@ -5,62 +5,104 @@ import (
...
+						}
+					default:
+						c.Logger().Error(err)
+						select {
+						case ch <- err:
+							c.Logger().Debug("Sent error to Writer channel")
+							return
+						case <-time.After(wsTimeout):
+							c.Logger().Debug("Timed out sending error to Writer channel")
+							return
```
[the remark this answers is not in the document]
Comment (Sarahtonein) [reply controllers/ws/state/stateWriter.go]: > Consider Using the new ErrorHandlers for this

Added :)

File: handlers/state.go
Comment (Balugrizzly) [reply handlers/state.go:1]: Not in the handlers

**Review by Balugrizzly — COMMENTED** [review]
Looks good 
lmk when you are ready for merge

File: handlers/state.go
Comment (Sarahtonein) [reply handlers/state.go:1]: > Why does this use an Array function signature? All other handlers don't

Have removed this, tested and getting same results listed in my initial PR comments :)

**Comment by Sarahtonein** [comment]
> Looks good lmk when you are ready for merge

Looks good to merge now if you are happy with changes :)

**Review by Balugrizzly — COMMENTED** [review]
Can you trim Logging?

**Comment by Balugrizzly** [comment]
Can you fix merge conflicts?

**Comment by Sarahtonein** [comment]
> Can you fix merge conflicts?

Merge conflicts resolved and logs trimmed

**Review by Balugrizzly — APPROVED** [review]
A

**Pull request merged**